### PR TITLE
Implement pkg/agent/core/ngt/service/option test 

### PR DESCRIPTION
**Pull request by kevindiu**
Signed-off-by: kevindiu <kevindiujp@gmail.com>

### Description:

This PR implements pkg/agent/core/ngt/service/option test.

### Environment:

- Go Version: 1.17
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/service/option.go [hunk 1]
```diff
@@ -133,6 +141,7 @@ func WithAutoIndexLength(l int) Option {
 	}
 }
 
+// WithInitialDelayMaxDuration returns the functional option to set the inital delay duration.
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply pkg/agent/core/ngt/service/option.go:144]:
```suggestion
// WithInitialDelayMaxDuration returns the functional option to set the initial delay duration.
```

File: pkg/agent/core/ngt/service/option.go [hunk 2]
```diff
@@ -91,6 +96,7 @@ func WithAutoIndexCheckDuration(dur string) Option {
 	}
 }
 
+// WithCache returns the functional option to set the cache.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/option.go:99]: is this correct?:thinking:

Comment (kevindiu) [reply pkg/agent/core/ngt/service/option.go:99]:
```suggestion
// WithAutoIndexDurationLimit returns the functional option to set the auto index duration limit.
```

**Review by hlts2 — DISMISSED** [review]

**Review by vankichi — DISMISSED** [review]

**Review by vankichi — APPROVED** [review]

**Review by hlts2 — DISMISSED** [review]

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]

**Pull request merged**